pkg/envoy/cds/cluster.go

```diff
@@ -41,19 +57,29 @@ func getUpstreamServiceCluster(downstreamIdentity identity.ServiceIdentity, upst
 	}
 
 	remoteCluster := &xds_cluster.Cluster{
-		Name:           clusterName,
-		ConnectTimeout: ptypes.DurationProto(clusterConnectTimeout),
-		TransportSocket: &xds_core.TransportSocket{
+		Name:                          upstreamSvc.String(),
+		ConnectTimeout:                ptypes.DurationProto(clusterConnectTimeout),
+		TypedExtensionProtocolOptions: HTTP2ProtocolOptions,
+	}
+
+	if !o.noTLS {
+		marshalledUpstreamTLSContext, err := ptypes.MarshalAny(
+			envoy.GetUpstreamTLSContext(downstreamIdentity, upstreamSvc))
+		if err != nil {
+			return nil, err
+		}
+		remoteCluster.TransportSocket = &xds_core.TransportSocket{
 			Name: wellknown.TransportSocketTls,
 			ConfigType: &xds_core.TransportSocket_TypedConfig{
 				TypedConfig: marshalledUpstreamTLSContext,
 			},
-		},
-		TypedExtensionProtocolOptions: HTTP2ProtocolOptions,
+		}
 	}
 
-	if cfg.IsPermissiveTrafficPolicyMode() {
+	if cfg.IsPermissiveTrafficPolicyMode() && !o.noPermissive {
```

Comment:
would let the caller set the permissive option instead based on `cfg.IsPermissiveTrafficPolicyMode()`, so we don't have similar conflicting conditionals around permissive mode here.

This logic should be as simple as 
```
if o.permissive {
 ...
}
```

Reply:
done!